core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -93,6 +93,13 @@ private TransactionOutPoint(Sha256Hash hash, long index, @Nullable Transaction f
         this.index = index;
         this.fromTx = fromTx;
         this.connectedOutput = connectedOutput;
+        if (fromTx != null) {
+            TransactionOutput outputFromTx = fromTx.getOutput(index);
+            Objects.requireNonNull(outputFromTx);
+            if (connectedOutput != null) {
+                checkArgument(connectedOutput.equals(outputFromTx), () -> "mismatched connected output");
+            }
+        }
```

Comment:
This looks good, but I would put the validation before the assignment to object members, as we typically do.

Reply:
Moved up.